src/workspace/resolver.rs

```diff
@@ -240,18 +242,135 @@ pub enum WorkspaceResolvePkgJsonFolderErrorKind {
 }
 
 #[derive(Debug)]
-pub struct WorkspaceResolver {
+struct CompilerOptionsRootDirsResolver<TSys: FsMetadata> {
+  root_dirs_from_root: Vec<Url>,
+  root_dirs_by_member: BTreeMap<Url, Option<Vec<Url>>>,
```

Comment:
Nitpick: Maybe we should just not have a member entry when its value is `None`? Or do we need to represent this as `None`? (if not, it would slightly reduce the number of comparisons)

Reply:
It makes a difference if you have members within members, to prevent cascading inheritance between them in some cases.

(We have cascading inheritance for import maps as a side-effect of using `scopes` but not for other fields.)